app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,99 @@
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+    private ArrayList subStrucCon = null;
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode node = translationUnit;
+        saveNodes(node);
+    }
+
+    /**Accesses all the nodes of the tree*/
+    private void saveNodes(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child : children) {
+
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    structHelper(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    saveAs("const", child);
+                }
+
+            }
+        }
+    }
+
+    private void saveAs(String t, IASTNode node) throws Exception{
+        CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) node;
+        CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
+        CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+
+        if (declarator != null || specifier != null || declaration != null) {
+
+            CPPASTName name = (CPPASTName) declarator.getName(); //get name
+            IToken typeQualifier = specifier.getSyntax(); //get typeQualifer type (iToken)
+            CPPASTName type = (CPPASTName) specifier.getName(); //get type
+
+            if (t.equals("struct")) {
+                StructContents content = new StructContents(name,type);
+                subStrucCon.add(content);
+            }
+
+            else if (t.equals("const")) {
+                CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+                if (init != null) {
+                    CPPASTLiteralExpression value = (CPPASTLiteralExpression) init.getInitializerClause();
+                    Contents contents = new Contents(name, value, typeQualifier, type);
+                    constRepo.put(name.getRawSignature(), contents);
+                }
+
+            }
+
+        }
+    }
+
+    private void structHelper(IASTDeclaration[] declarations, String name) throws Exception {
+        subStrucCon = new ArrayList();
```

Comment:
I'd suggest minimizing the amount of state kept in instance variables where not necessary. Here, `subStructCon` seems mainly being used to share data between `structHelper` and `saveAs`, instead of being useful persistent state. Instead, consider having `saveAs` (or the refactored version of that) return either the filled ArrayList, or have it return an element to add to an ArrayList created solely within `stuctHelper`